scripts/benchmark.py

```diff
@@ -0,0 +1,475 @@
+        return Benchmark(b[1], test=Test(b[2], b[3]), realruns=real, userruns=user)
+
+    def store(self, benchmark, rev=None):
+        """stores a (run/executed) Benchmark in the database. if rev is None, git revision is retrieved from git"""
+        ex = benchmark.test.executable
+        infile = benchmark.test.infile
+        timestamp = time.time()
+        date = timestamp
+        if rev is None:
+            rev, date = git_revision()
+        load = os.getloadavg()[0]
+
+        c = self.conn.cursor()
+        c.execute('INSERT INTO benchmarks (name,executable,input_file,timestamp,revision,date,load) VALUES (?,?,?,?,?,?,?)',
+                (benchmark.name, ex, infile, timestamp, rev, date, load))
+        self.conn.commit()
+
+        c.execute('SELECT max(id) from benchmarks')
+        bench_id = int(c.fetchone()[0])
+
+        i = 0
+        for real, user, perflog in zip(benchmark.realruns, benchmark.userruns, benchmark.perflogruns):
+            c.execute('INSERT INTO timings (benchmark_id, run, realtime_secs, usertime_secs) VALUES (?,?,?,?)', (bench_id, i, real, user))
+            i += 1
+            for entry in perflog:
+                cat, subcat, nruns, selftime, cumtime = entry
+                c.execute('INSERT INTO perflog (benchmark_id, run, field, subfield, exec_count, self_time_secs, cum_time_secs) VALUES (?,?,?,?,?,?,?)',
+                        (bench_id, i, cat, subcat, nruns, selftime, cumtime))
+
+        return bench_id
+
+    def close(self):
+        self.conn.commit()
+        self.conn.close()
+
+def git_revision(dir='.'):
+    # return hash and (unix secs since epoch) date
+    cmd = ['git', 'log', '--date', 'unix', '--pretty=format:%H %ad', '-n', '1']
+    p = subprocess.Popen(cmd, stdout=subprocess.PIPE, cwd=dir)
+    stdout, stderr = p.communicate()
+    if p.returncode != 0:
+        raise RuntimeError('failed to retrieve git revision')
+    commit = stdout.strip().split(' ')[0]
+    date = int(stdout.strip().split(' ')[1])
+    return commit, date
+
+class BenchEntry:
+    def __init__(self, name, executable, input_file, args):
+        self.name = name
+        self.executable = executable
+        self.infile = input_file
+        self.args = args.split(' ')
+
+def read_benchmarks(benchlist):
+    if not os.path.exists(benchlist):
+        raise ValueError('benchmark list file "{}" does not exist'.format(benchlist))
+
+    benches = []
+    root = readInputFile(benchlist).children['benchmarks']
+    for benchname in root.children_list:
+        child = root.children[benchname]
+        ex = child.params['binary'].strip()
+        infile = child.params['infile'].strip()
+        args = ''
+        if 'args' in child.params:
+            args = child.params['args']
+        benches.append(BenchEntry(benchname, ex, infile, args))
+    return benches
+
+if __name__ == '__main__':
+    find_moose_python()
+    from FactorySystem.ParseGetPot import readInputFile
```

Comment:
Because we need to add MOOSE_DIR/python to the list of dirs python hunts for modules in since this is a script and doesn't live in the python directory.